plonky2/src/gates/selectors.rs

```diff
@@ -0,0 +1,82 @@
+use plonky2_field::extension_field::Extendable;
+use plonky2_field::polynomial::PolynomialValues;
+
+use crate::gates::gate::{GateInstance, GateRef};
+use crate::hash::hash_types::RichField;
+
+/// Placeholder value to indicate that a gate doesn't use a selector polynomial.
+pub(crate) const UNUSED_SELECTOR: usize = u32::MAX as usize;
+
+#[derive(Debug, Clone)]
+pub(crate) struct SelectorsInfo {
+    pub(crate) selector_indices: Vec<usize>,
+    pub(crate) groups: Vec<(usize, usize)>,
```

Comment:
nit: I think `Vec<Range<usize>>` may be nice - it would be a bit more self-commenting, and would slightly simplify `group` (using `contains`) and the `compute_filter`s (no literal range)